signac/core/jsoncollection.py

```diff
@@ -6,59 +6,97 @@
 This implements the JSON-backend for SyncedCollection API by
 implementing sync and load methods.
 """
-
 import os
 import json
 import errno
 import uuid
 
 from .synced_collection import SyncedCollection
+from .buffered_collection import BufferedSyncedCollection
 from .syncedattrdict import SyncedAttrDict
 from .synced_list import SyncedList
+from .buffered_collection import _get_filemetadata
+
+
+def get_namespace(class_name):
+    """Generate namespace for classname"""
+    return uuid.uuid5(uuid.NAMESPACE_URL, 'signac::'+class_name)
```

Comment:
I'm not sure this name is going to be a valid URL. You need to generate a URL that includes the class name and backend kwargs in the same "name."
```suggestion
    return uuid.uuid5(uuid.NAMESPACE_URL, 'signac://'+class_name)
```